### ISSUE TITLE
Date.RelativeFormatStyle is rounding up

### ISSUE THREAD

**Issue by BrentMifsud** [apple/swift-foundation#774]
We recently switched over to using FormatStyles instead of Dateformatters.

I noticed some odd behavior with the relative date format style.

```swift
import Foundation

let date = Calendar.autoupdatingCurrent.date(byAdding: .second, value: 90, to: .now)!
let date2 = Calendar.autoupdatingCurrent.date(byAdding: .second, value: 89, to: .now)!
date.formatted(.relative(presentation: .numeric, unitsStyle: .wide)) // in 2 minutes
date2.formatted(.relative(presentation: .numeric, unitsStyle: .wide)) // in 1 minute
```

Is this the intended behaviour?

Using legacy RelativeDateFormatters, it says "in 1 minute" until you actually reach the 2 min mark

**Comment by itingliu** [comment]
It's probably more correct to say it's "rounding to the nearest minute" instead of "rounding up". 89 seconds is 1.48 minutes, so it's displayed as "in 1 minute", whereas "90 seconds" is 1.5 minutes, so it becomes "in 2 minutes"  (see [here](https://github.com/apple/swift-foundation/blob/008613336a15bbc009e00234a8e2eae1bfb1ed12/Sources/FoundationInternationalization/Formatting/Date/Date%2BRelativeFormatStyle.swift#L204C29-L204C58) for implementation 

And yes it is intentional. We had some feedback about the current behaviour of `RelativeDateFormatters` being undesired, so we made this choice intentionally. I think it would be useful to expose an API to customise this though.

**Comment by BrentMifsud** [comment]
> It's probably more correct to say it's "rounding to the nearest minute" instead of "rounding up". 89 seconds is 1.48 minutes, so it's displayed as "in 1 minute", whereas "90 seconds" is 1.5 minutes, so it becomes "in 2 minutes" (see [here](https://github.com/apple/swift-foundation/blob/008613336a15bbc009e00234a8e2eae1bfb1ed12/Sources/FoundationInternationalization/Formatting/Date/Date%2BRelativeFormatStyle.swift#L204C29-L204C58) for implementation
> 
> And yes it is intentional. We had some feedback about the current behaviour of `RelativeDateFormatters` being undesired, so we made this choice intentionally. I think it would be useful to expose an API to customise this though.

Hmm, I did notice that this rounding also occurs with larger units of time as well (weeks, years, etc...). Which definitely starts to feel more incorrect.

```
let calendar = Calendar.autoupdatingCurrent

let formatStyle = Date.RelativeFormatStyle(presentation: .numeric, unitsStyle: .wide, locale: .autoupdatingCurrent, calendar: calendar, capitalizationContext: .middleOfSentence)

let today = Date()
let oneWeekFromNow = calendar.date(byAdding: .weekOfYear, value: 1, to: today)
let nineDaysFromNow = calendar.date(byAdding: .day, value: 9, to: today)
let oneYearFromNow = calendar.date(byAdding: .year, value: 1, to: today)
let oneYearSixMonthsFromNow = calendar.date(byAdding: .month, value: 18, to: today)
oneWeekFromNow!.formatted(formatStyle) // in 1 week
nineDaysFromNow!.formatted(formatStyle) // in 2 weeks
oneYearFromNow!.formatted(formatStyle) // in 1 year
oneYearSixMonthsFromNow!.formatted(formatStyle) // in 2 years
```

**Comment by itingliu** [comment]
>I did notice that this rounding also occurs with larger units of time as well (weeks, years, etc...). 

You're right. When I said "it's rounding to the nearest minute" previously, I should have said that it's "rounding to the nearest **component**", and is not particular to "minute".
 
>Would definitely be nice to be able to configure the rounding options, as well as the date components we want to show.

We recently added the [`allowFields` property](https://github.com/apple/swift-foundation/blob/aee18e4289f4c99cd8315fd507a9da3846dcf5ce/Sources/FoundationInternationalization/Formatting/Date/Date%2BRelativeFormatStyle.swift#L99) to let you choose which components to show. There are some examples in [this test](https://github.com/apple/swift-foundation/blob/aee18e4289f4c99cd8315fd507a9da3846dcf5ce/Tests/FoundationInternationalizationTests/Formatting/DateRelativeFormatStyleTests.swift#L332). Let us know if that's a good start!

**Comment by BrentMifsud** [comment]
> >I did notice that this rounding also occurs with larger units of time as well (weeks, years, etc...). 
> 
> 
> 
> You're right. When I said "it's rounding to the nearest minute" previously, I should have said that it's "rounding to the nearest **component**", and is not particular to "minute".
> 
>  
> 
> >Would definitely be nice to be able to configure the rounding options, as well as the date components we want to show.
> 
> 
> 
> We recently added the [`allowFields` property](https://github.com/apple/swift-foundation/blob/aee18e4289f4c99cd8315fd507a9da3846dcf5ce/Sources/FoundationInternationalization/Formatting/Date/Date%2BRelativeFormatStyle.swift#L99) to let you choose which components to show. There are some examples in [this test](https://github.com/apple/swift-foundation/blob/aee18e4289f4c99cd8315fd507a9da3846dcf5ce/Tests/FoundationInternationalizationTests/Formatting/DateRelativeFormatStyleTests.swift#L332). Let us know if that's a good start!

Unfortunately the new AllowedFields aren't backported to earlier iOS versions. So I can't use them. But will test it out in a playground to see if it does what I need.

**Comment by EngOmarElsayed** [comment]
if this issue is still going can I work on it ?
